packages/cli/src/deploy/deploy.ts

```diff
@@ -28,6 +29,7 @@ type DeployOptions<configInput extends ConfigInput> = {
    * not have a deterministic address.
    */
   deployerAddress?: Hex;
+  deployAsProxy: boolean;
```

Comment:
nit, should we call this something like `withWorldProxy`? or `deployWorldAsProxy` but that feels a bit too verbose 🤔